substrate/frame/election-provider-multi-block/src/signed/tests.rs

```diff
@@ -324,10 +679,190 @@ mod solution_data_provider {
 	}
 
 	#[test]
-	fn get_page_works() {
+	fn get_score_returns_entry_with_highest_sum_stake_given_entries_with_equal_minimal_stakes() {
 		ExtBuilder::default().build_and_execute(|| {
 			roll_to_phase(Phase::Signed);
+
 			assert_eq!(<SignedPallet as SolutionDataProvider>::get_score(), None);
+
+			let higher_score =
+				ElectionScore { minimal_stake: 40, sum_stake: 10, sum_stake_squared: 0 };
+			assert_ok!(SignedPallet::register(RuntimeOrigin::signed(40), higher_score));
+
+			let score = ElectionScore { minimal_stake: 40, sum_stake: 5, sum_stake_squared: 0 };
+			assert_ok!(SignedPallet::register(RuntimeOrigin::signed(30), score));
+
+			assert_eq!(<SignedPallet as SolutionDataProvider>::get_score(), Some(higher_score));
+		})
+	}
+
+	#[test]
+	fn get_score_returns_entry_with_lowest_sum_stake_squared_given_entries_with_equal_minimal_stakes_and_sum_stakes(
+	) {
+		ExtBuilder::default().build_and_execute(|| {
+			roll_to_phase(Phase::Signed);
+
+			assert_eq!(<SignedPallet as SolutionDataProvider>::get_score(), None);
+
+			let higher_score =
+				ElectionScore { minimal_stake: 40, sum_stake: 10, sum_stake_squared: 2 };
+			assert_ok!(SignedPallet::register(RuntimeOrigin::signed(40), higher_score));
+
+			let score = ElectionScore { minimal_stake: 40, sum_stake: 10, sum_stake_squared: 5 };
+			assert_ok!(SignedPallet::register(RuntimeOrigin::signed(30), score));
+
+			assert_eq!(<SignedPallet as SolutionDataProvider>::get_score(), Some(higher_score));
+		})
+	}
+
+	#[test]
+	fn get_paged_solution_returns_previously_submitted_page() {
+		ExtBuilder::default().build_and_execute(|| {
+			let origin = RuntimeOrigin::signed(99);
+			roll_to_phase(Phase::Signed);
+
+			assert_ok!(SignedPallet::register(origin.clone(), Default::default()));
+			assert_ok!(SignedPallet::submit_page(origin, 0, Some(Default::default())));
+
+			assert_ne!(<SignedPallet as SolutionDataProvider>::get_paged_solution(0), None)
+		})
+	}
+
+	#[test]
+	fn get_paged_solution_returns_none_given_invalid_page_index() {
+		ExtBuilder::default().build_and_execute(|| {
+			let origin = RuntimeOrigin::signed(99);
+			roll_to_phase(Phase::Signed);
+
+			assert_ok!(SignedPallet::register(origin.clone(), Default::default()));
+			assert_ok!(SignedPallet::submit_page(origin, 0, Some(Default::default())));
+
+			assert_eq!(<SignedPallet as SolutionDataProvider>::get_paged_solution(12345), None)
+		})
+	}
+
+	#[test]
```

Comment:
This test could be merged with the above test. Calling it `get_wrong_solution_page_works` and then testing all the cases where the None is returned due to bad page index is a good call !

Reply:
As above. I'm not in favour for merged scenarios so let's discuss that bigger topic first